cpp/tests/io/orc_test.cpp

```diff
@@ -2427,4 +2428,25 @@ INSTANTIATE_TEST_CASE_P(Host,
                                                              cudf::io::compression_type::SNAPPY,
                                                              cudf::io::compression_type::ZSTD)));
 
+TEST_F(OrcWriterTest, DISABLED_SizeTypeOverflow)
+{
+  constexpr auto num_rows = std::numeric_limits<cudf::size_type>::max();
+
+  auto const val = cudf::numeric_scalar<bool>(true);
+  auto const col = cudf::make_column_from_scalar(val, num_rows);
+  ASSERT_EQ(col->size(), num_rows);
+
+  std::vector<char> buffer;
+  buffer.reserve(num_rows);
```

Comment:
Not sure if this reserve will help (not complaining just pointing out 😄), will it?

Reply:
It might; we do keep resizing the sink vector as we write into it. And we definitely don't write in a single step.